webapp/packages/plugin-connections/src/DatabaseAuthDialog/DatabaseSecretAuthDialog/DatabaseSecretAuthDialog.tsx

```diff
@@ -0,0 +1,90 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observable } from 'mobx';
+import { observer } from 'mobx-react-lite';
+
+import {
+  CommonDialogBody,
+  CommonDialogFooter,
+  ExceptionMessage,
+  Group,
+  ItemList,
+  ListItem,
+  ListItemName,
+  Loader,
+  useObservableRef,
+  useResource,
+  useS,
+} from '@cloudbeaver/core-blocks';
+import { ConnectionInfoResource, type IConnectionInfoParams } from '@cloudbeaver/core-connections';
+
+import style from './DatabaseSecretAuthDialog.m.css';
+
+interface Props {
+  connectionKey: IConnectionInfoParams;
+  onLogin?: () => void;
+}
+
+export const DatabaseSecretAuthDialog = observer<Props>(function DatabaseSecretAuthDialog({ connectionKey, onLogin }) {
+  const styles = useS(style);
+  const connectionInfoLoader = useResource(DatabaseSecretAuthDialog, ConnectionInfoResource, {
+    key: connectionKey,
+    includes: ['includeAuthNeeded', 'includeSharedSecrets', 'includeNetworkHandlersConfig', 'includeCredentialsSaved'],
+  });
+  const state = useObservableRef(
+    () => ({
+      exception: null as Error | null,
+      authenticating: false,
+    }),
+    {
+      exception: observable.ref,
+      authenticating: observable.ref,
+    },
+    false,
+  );
+  const secrets = connectionInfoLoader.data?.sharedSecrets || [];
+
+  async function handleSecretSelect(secretId: string) {
+    try {
+      state.authenticating = true;
+      await connectionInfoLoader.resource.init({
+        ...connectionKey,
+        selectedSecretId: secretId,
+      });
+      state.exception = null;
```

Comment:
probably we want to set exception to null before we try to init connection?

Reply:
actually it doesn't matter here because we will hide exception message while connecting